### ISSUE TITLE
Walk/Walk2, Noise and S&H crash Rack [GCC v10.2.0/Win 10]

### ISSUE THREAD

**Issue by SteveRussell33** [jeremywen/JW-Modules#59]
After building master branch against latest Rack source, selecting the above modules from the browser causes a crash.
```
Walk
[94.029 fatal src/main.cpp:45] Fatal signal 22. Stack trace:
36: ZN4rack10appDestroyEv 0x456d75
35: raise 0xc985ac60
34: abort 0xc985f270
33: ZN9__gnu_cxx27__verbose_terminate_handlerEv 0x6fc5f170
32: ZN10__cxxabiv111__terminateEPFvvE 0x6fc581b0
31: ZSt9terminatev 0x6fd302b0
30: _cxa_throw 0x6fd39080
29: ZSt21__throw_runtime_errorPKc 0x6fd3a640
28: ZNSt13random_device7_M_initERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE 0x6fcf10f0
27: ZN8bogaudio3dsp5SeedsC1Ev 0xc46146a
26: ZN8bogaudio3dsp5Seeds4nextEv 0xc46190e
25: ZN8bogaudio4WalkC1Ev 0xc5265a0
24: ZZN4rack11createModelIN8bogaudio4WalkE10WalkWidgetEEPNS_6plugin5ModelERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEN6TModel18createModuleWidgetEv 0xc587f00
23: ZN4rack3app11AudioWidget12setAudioPortEPNS_5audio4PortE 0x49869c
22: ZN4rack3app8ModelBox8onButtonERKNS_5event6ButtonE 0x75ec00
21: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
20: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
19: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
18: ZN4rack2ui12ScrollWidget8onButtonERKNS_5event6ButtonE 0x4a4444
17: ZN4rack6widget12OpaqueWidget8onButtonERKNS_5event6ButtonE 0x773d20
16: ZN4rack3app14BrowserOverlay8onButtonERKNS_5event6ButtonE 0x7550a0
15: ZN4rack6widget12OpaqueWidget8onButtonERKNS_5event6ButtonE 0x773d20
14: ZN4rack5event5State12handleButtonENS_4math3VecEiii 0x458008
13: ZN4rack7updater17isUpdateAvailableEv 0x459a8e
12: glfwPlatformUnlockMutex 0x4d7e2e
11: CallWindowProcW 0xc87c5850
10: CallWindowProcW 0xc87c5850
9: glPushClientAttrib 0xb53f3770
8: CallWindowProcW 0xc87c5850
7: DispatchMessageW 0xc87c5410
6: glfwPlatformPollEvents 0x4db4e6
5: ZN4rack6Window3runEv 0x459e86
4: main 0x7838a0
3: main 0x7838a0
2: main 0x7838a0
1: BaseThreadInitThunk 0xc9607bc0
0: RtlUserThreadStart 0xca76ce30

Walk2
[97.682 fatal src/main.cpp:45] Fatal signal 22. Stack trace:
36: ZN4rack10appDestroyEv 0x456d75
35: raise 0xc985ac60
34: abort 0xc985f270
33: ZN9__gnu_cxx27__verbose_terminate_handlerEv 0x6fc5f170
32: ZN10__cxxabiv111__terminateEPFvvE 0x6fc581b0
31: ZSt9terminatev 0x6fd302b0
30: _cxa_throw 0x6fd39080
29: ZSt21__throw_runtime_errorPKc 0x6fd3a640
28: ZNSt13random_device7_M_initERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE 0x6fcf10f0
27: ZN8bogaudio3dsp5SeedsC1Ev 0xc45146a
26: ZN8bogaudio3dsp5Seeds4nextEv 0xc45190e
25: ZN8bogaudio5Walk2C1Ev 0xc52bce0
24: ZZN4rack11createModelIN8bogaudio5Walk2E11Walk2WidgetEEPNS_6plugin5ModelERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEN6TModel18createModuleWidgetEv 0xc57a120
23: ZN4rack3app11AudioWidget12setAudioPortEPNS_5audio4PortE 0x49869c
22: ZN4rack3app8ModelBox8onButtonERKNS_5event6ButtonE 0x75ec00
21: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
20: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
19: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
18: ZN4rack2ui12ScrollWidget8onButtonERKNS_5event6ButtonE 0x4a4444
17: ZN4rack6widget12OpaqueWidget8onButtonERKNS_5event6ButtonE 0x773d20
16: ZN4rack3app14BrowserOverlay8onButtonERKNS_5event6ButtonE 0x7550a0
15: ZN4rack6widget12OpaqueWidget8onButtonERKNS_5event6ButtonE 0x773d20
14: ZN4rack5event5State12handleButtonENS_4math3VecEiii 0x458008
13: ZN4rack7updater17isUpdateAvailableEv 0x459a8e
12: glfwPlatformUnlockMutex 0x4d7e2e
11: CallWindowProcW 0xc87c5850
10: CallWindowProcW 0xc87c5850
9: glPushClientAttrib 0xb53f3770
8: CallWindowProcW 0xc87c5850
7: DispatchMessageW 0xc87c5410
6: glfwPlatformPollEvents 0x4db4e6
5: ZN4rack6Window3runEv 0x459e86
4: main 0x7838a0
3: main 0x7838a0
2: main 0x7838a0
1: BaseThreadInitThunk 0xc9607bc0
0: RtlUserThreadStart 0xca76ce30

Noise
[86.497 fatal src/main.cpp:45] Fatal signal 22. Stack trace:
36: ZN4rack10appDestroyEv 0x456d75
35: raise 0xc985ac60
34: abort 0xc985f270
33: ZN9__gnu_cxx27__verbose_terminate_handlerEv 0x6fc5f170
32: ZN10__cxxabiv111__terminateEPFvvE 0x6fc581b0
31: ZSt9terminatev 0x6fd302b0
30: _cxa_throw 0x6fd39080
29: ZSt21__throw_runtime_errorPKc 0x6fd3a640
28: ZNSt13random_device7_M_initERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE 0x6fcf10f0
27: ZN8bogaudio3dsp5SeedsC1Ev 0xc44146a
26: ZN8bogaudio3dsp5Seeds4nextEv 0xc44190e
25: ZN8bogaudio5NoiseC1Ev 0xc511db0
24: ZZN4rack11createModelIN8bogaudio5NoiseE11NoiseWidgetEEPNS_6plugin5ModelERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEN6TModel18createModuleWidgetEv 0xc568ec0
23: ZN4rack3app11AudioWidget12setAudioPortEPNS_5audio4PortE 0x49869c
22: ZN4rack3app8ModelBox8onButtonERKNS_5event6ButtonE 0x75ec00
21: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
20: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
19: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
18: ZN4rack2ui12ScrollWidget8onButtonERKNS_5event6ButtonE 0x4a4444
17: ZN4rack6widget12OpaqueWidget8onButtonERKNS_5event6ButtonE 0x773d20
16: ZN4rack3app14BrowserOverlay8onButtonERKNS_5event6ButtonE 0x7550a0
15: ZN4rack6widget12OpaqueWidget8onButtonERKNS_5event6ButtonE 0x773d20
14: ZN4rack5event5State12handleButtonENS_4math3VecEiii 0x458008
13: ZN4rack7updater17isUpdateAvailableEv 0x459a8e
12: glfwPlatformUnlockMutex 0x4d7e2e
11: CallWindowProcW 0xc87c5850
10: CallWindowProcW 0xc87c5850
9: glPushClientAttrib 0xb53f3770
8: CallWindowProcW 0xc87c5850
7: DispatchMessageW 0xc87c5410
6: glfwPlatformPollEvents 0x4db4e6
5: ZN4rack6Window3runEv 0x459e86
4: main 0x7838a0
3: main 0x7838a0
2: main 0x7838a0
1: BaseThreadInitThunk 0xc9607bc0
0: RtlUserThreadStart 0xca76ce30

S&H
[27.824 fatal src/main.cpp:45] Fatal signal 22. Stack trace:
36: ZN4rack10appDestroyEv 0x456d75
35: raise 0xc985ac60
34: abort 0xc985f270
33: ZN9__gnu_cxx27__verbose_terminate_handlerEv 0x6fc5f170
32: ZN10__cxxabiv111__terminateEPFvvE 0x6fc581b0
31: ZSt9terminatev 0x6fd302b0
30: _cxa_throw 0x6fd39080
29: ZSt21__throw_runtime_errorPKc 0x6fd3a640
28: ZNSt13random_device7_M_initERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE 0x6fcf10f0
27: ZN8bogaudio3dsp5SeedsC1Ev 0xc48146a
26: ZN8bogaudio3dsp5Seeds4nextEv 0xc48190e
25: ZN8bogaudio10SampleHoldC1Ev 0xc4f84c0
24: ZZN4rack11createModelIN8bogaudio10SampleHoldE16SampleHoldWidgetEEPNS_6plugin5ModelERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEN6TModel18createModuleWidgetEv 0xc59e740
23: ZN4rack3app11AudioWidget12setAudioPortEPNS_5audio4PortE 0x49869c
22: ZN4rack3app8ModelBox8onButtonERKNS_5event6ButtonE 0x75ec00
21: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
20: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
19: ZN4rack6widget6Widget8onButtonERKNS_5event6ButtonE 0x774740
18: ZN4rack2ui12ScrollWidget8onButtonERKNS_5event6ButtonE 0x4a4444
17: ZN4rack6widget12OpaqueWidget8onButtonERKNS_5event6ButtonE 0x773d20
16: ZN4rack3app14BrowserOverlay8onButtonERKNS_5event6ButtonE 0x7550a0
15: ZN4rack6widget12OpaqueWidget8onButtonERKNS_5event6ButtonE 0x773d20
14: ZN4rack5event5State12handleButtonENS_4math3VecEiii 0x458008
13: ZN4rack7updater17isUpdateAvailableEv 0x459a8e
12: glfwPlatformUnlockMutex 0x4d7e2e
11: CallWindowProcW 0xc87c5850
10: CallWindowProcW 0xc87c5850
9: glPushClientAttrib 0xb53f3770
8: CallWindowProcW 0xc87c5850
7: DispatchMessageW 0xc87c5410
6: glfwPlatformPollEvents 0x4db4e6
5: ZN4rack6Window3runEv 0x459e86
4: main 0x7838a0
3: main 0x7838a0
2: main 0x7838a0
1: BaseThreadInitThunk 0xc9607bc0
0: RtlUserThreadStart 0xca76ce30
```
I've narrowed it down to the fact that these modules use ``dsp/noise.hpp`` (std::random_device?)

**Comment by SteveRussell33** [comment]
The notes here https://en.cppreference.com/w/cpp/numeric/random/random_device may give some clue, as I'm using the newest GCC version.

`A notable implementation where std::random_device is deterministic is old versions of MinGW (bug 338, fixed since GCC 9.2), although replacement implementations exist, such as mingw-std-random_device.`

**Comment by SteveRussell33** [comment]
Opps, just realised wrong repo. Apologies.
